services/src/main/java/org/jboss/windup/web/services/rest/WindupEndpointImpl.java

```diff
@@ -55,18 +65,49 @@
     private Queue executorQueue;
 
     /**
+     * To avoid problems with passing detached entities,
+     * load related entities during cloning
+     */
+    protected AnalysisContext cloneExecution(AnalysisContext originalContext)
+    {
+        Set<Package> allPackages = new HashSet<>();
+
+        allPackages.addAll(originalContext.getIncludePackages());
+        allPackages.addAll(originalContext.getExcludePackages());
+
+        List<Long> packageIds = allPackages.stream().map(aPackage -> aPackage.getId())
+                .collect(Collectors.toList());
+
+        AnalysisContext clone = originalContext.clone();
```

Comment:
@klinki - Arguably, if analysis context doesn't match then those other items may not match either. Any of them could have been updated. :)

If we care about this, I think that an MVCC (optimistic locking approach) might be simpler. Just throw if the user has passed an outdated AnalysisContext object, since we do have a version field.